apps/docs/components/Utils/CopyButton/CopyButton.tsx

```diff
@@ -0,0 +1,29 @@
+"use client";
+
+import React from "react";
+import "./copyButton.css";
+
+export const CopyButton = ({ text }: { text: string }) => {
+    const [isCopied, setIsCopied] = React.useState(false);
+
+    const copy = async () => {
+        await navigator.clipboard.writeText(text);
+        setIsCopied(true);
+
+        setTimeout(() => {
+            setIsCopied(false);
+        }, 10000);
+    };
+
+    return (
+        <button disabled={isCopied} onClick={copy} className="hd-copy-button">
```

Comment:
You should use [react-aria](https://react-spectrum.adobe.com/react-aria/useButton.html) `useButton` for this component

Reply:
As I have a branch for converting Radix to React-Aria I think this could be done in this other branch as we would otherwise will live with both dependencies which is not ideal.